geode-core/src/main/java/org/apache/geode/internal/cache/TXState.java

```diff
@@ -1936,10 +1936,10 @@ public boolean isRealDealLocal() {
 
   @Override
   public InternalDistributedMember getOriginatingMember() {
-    /*
-     * State will never fwd on to other nodes so this is not relevant
-     */
-    return null;
+    // During set operations we need to forward it to other nodes to avoid
+    // wrong transaction is masqueraded if the transaction is on behalf of a client.
+    // This needs to be set to the clients member id if the client originated the tx.
+    return ((TXStateProxyImpl) proxy).onBehalfOfClientMember;
```

Comment:
1. change this field access to instead call a method
2. get rid of the cast by adding that method to TXStateProxy